This notebook aims to identify spam areas in Wikidata.

In [1]:
import pandas as pd

In [2]:
# read predownloaded dataset. This is a sample of 44107 cities from wikidata with their population
df_wiki = pd.read_csv('sample_data/query_cities_population_entityid.csv')

# Sort dataframe in ascending order by population
df_wiki.sort_values(by=['population'], ascending=True, inplace=True)
df_wiki.head()

city        cityLabel population
41018  http://www.wikidata.org/entity/Q20541864        Lousadela          0
15985     http://www.wikidata.org/entity/Q48791       Tower City          0
40007  http://www.wikidata.org/entity/Q12381764        Q12381764          0
33214   http://www.wikidata.org/entity/Q1991598  South Park View          0
4335    http://www.wikidata.org/entity/Q1475379           Rineia          0

Cities with 0 population shouldn't exist. They are probably spam.
Even if the city is real, 0 population means the person who added the city probably took no efforts to fill in credible data, so the accuracy of the data is questionable.

In [9]:
df_wiki[(df_wiki['population'] < '100') & (df_wiki['population'] > '0')]

city  \
41042   http://www.wikidata.org/entity/Q20547092   
40008   http://www.wikidata.org/entity/Q12381764   
18902     http://www.wikidata.org/entity/Q729570   
41097  http://www.wikidata.org/entity/Q112321797   
39463   http://www.wikidata.org/entity/Q12382422   
37178    http://www.wikidata.org/entity/Q7399605   
44044   http://www.wikidata.org/entity/Q61379272   
40938   http://www.wikidata.org/entity/Q12397264   
37184    http://www.wikidata.org/entity/Q9097495   
41160  http://www.wikidata.org/entity/Q113513568   
40977   http://www.wikidata.org/entity/Q20543211   
37303     http://www.wikidata.org/entity/Q132830   
8771      http://www.wikidata.org/entity/Q132830   
39450     http://www.wikidata.org/entity/Q132830   
39841   http://www.wikidata.org/entity/Q12384976   
19125     http://www.wikidata.org/entity/Q651693   
40855   http://www.wikidata.org/entity/Q12386028   
40697   http://www.wikidata.org/entity/Q12395836   
25598    http://www.wikidata.org/entity/Q1519793   
41103  http://www.wikidata.org/entity/Q112321931   
40512   http://www.wikidata.org/entity/Q12388343   
41116  http://www.wikidata.org/entity/Q112669650   
41112  http://www.wikidata.org/entity/Q112669566   
37181    http://www.wikidata.org/entity/Q8197710   
30475    http://www.wikidata.org/entity/Q2167397   
39641   http://www.wikidata.org/entity/Q12387224   
39145   http://www.wikidata.org/entity/Q12382102   
41094  http://www.wikidata.org/entity/Q112321720   
41121  http://www.wikidata.org/entity/Q113513494   
37990   http://www.wikidata.org/entity/Q12380320   
41133  http://www.wikidata.org/entity/Q113513514   
30758    http://www.wikidata.org/entity/Q2069603   
40410   http://www.wikidata.org/entity/Q12394784   
41064  http://www.wikidata.org/entity/Q110282083   
28274    http://www.wikidata.org/entity/Q1907413   
33086    http://www.wikidata.org/entity/Q2316274   
31581    http://www.wikidata.org/entity/Q2096482   
30061    http://www.wikidata.org/entity/Q2154539   
44029   http://www.wikidata.org/entity/Q31283161   
32942    http://www.wikidata.org/entity/Q2230488   

                            cityLabel population  
41042                       Q20547092          1  
40008                       Q12381764          1  
18902                            Ruso          1  
41097                       O Palacio          1  
39463                       Q12382422          1  
37178                           O Vao          1  
44044  Cartelos, Cartelos, Carballedo          1  
40938                            Pazo          1  
37184                         Zaparín          1  
41160                    Souto de Rei          1  
40977                       Q20543211          1  
37303                          Douala       1.49  
8771                           Douala       1.49  
39450                          Douala       1.49  
39841                       Q12384976         10  
19125                        Venturia         10  
40855                       Q12386028         10  
40697                       O Robledo         10  
25598                         Clayton         10  
41103                         Xagrade         10  
40512                       Q12388343         10  
41116                          Parada         10  
41112                    O Figueiredo         10  
37181                        Q8197710         10  
30475                          Overly         10  
39641                            Ducí         10  
39145                       Q12382102         10  
41094                        Marcelle         10  
41121                            Alén         10  
37990                       Q12380320         10  
41133                A Porta do Prado         10  
30758                          Bergen         10  
40410                       Q12394784         10  
41064                      Q110282083         10  
28274                            Loma         10  
33086                           Wales         10  
31581                   

Again, some cities claim to have 1.49 living humans, which is highly questionable. Non-integer population values are likely to be spam.